src/Interpreters/Aggregator.cpp

```diff
@@ -1878,6 +1878,70 @@ Block Aggregator::mergeAndConvertOneBucketToBlock(
     return block;
 }
 
+template <typename Method>
+void Aggregator::mergeSingleLevelDataImplFixedMap(
+    ManyAggregatedDataVariants & non_empty_data,
+    const UInt32 worker_id,
+    const UInt32 total_worker,
+    std::atomic<bool> & is_cancelled) const
+{
+    AggregatedDataVariantsPtr & res = non_empty_data[0];
```

Comment:
This needs to be fixed. tsan [error](https://s3.amazonaws.com/clickhouse-test-reports/PRs/87366/79b9ad8d38032f021cec92bf4740c72a6ffab18b//stateless_tests_amd_tsan_parallel_2_2/stderr.log) is caused by different worker thread updating the same `many_data_variant[0]`.

Probably also causing some test like 00376_shard_group_uniq_array_of_int_array [fails](https://s3.amazonaws.com/clickhouse-test-reports/json.html?PR=87366&sha=79b9ad8d38032f021cec92bf4740c72a6ffab18b&name_0=PR&name_1=Stateless+tests+%28arm_binary%2C+parallel%29&name_2=Tests)..

Maybe we can let each worker merge to a local ad-hoc fixed hash table first. Once all worker finishes, merge them together in a single thread into data[0].